instrumentation/hibernate/hibernate-3.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/QueryInstrumentation.java

```diff
@@ -50,49 +50,33 @@ public void transform(TypeTransformer transformer) {
   public static class QueryMethodAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void startMethod(
-        @Advice.This Query query,
-        @Advice.Local("otelCallDepth") CallDepth callDepth,
-        @Advice.Local("otelHibernateOperation") HibernateOperation hibernateOperation,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
+    public static HibernateOperationScope startMethod(@Advice.This Query query) {
 
-      callDepth = CallDepth.forClass(HibernateOperation.class);
+      CallDepth callDepth = CallDepth.forClass(HibernateOperation.class);
       if (callDepth.getAndIncrement() > 0) {
-        return;
+        return HibernateOperationScope.wrapCallDepth(callDepth);
```

Comment:
IIRC, hibernate does a lot of method overload delegation which leads to fairly heavy usage of the CallDepth checks, it may be worth avoiding these allocations and conditionally returning CallDepth or HibernateOperationScope

Reply:
I've switched to less wrapping in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11553/commits/cd99412ef0739cda1482bdfbbd1d8a571eb4f6f4, should be better now.

Is there a usual benchmark to measure extra allocation created by the instrumentation/sdk ?